### PR TITLE
[10.0][MIG] printer_tray

### PR DESCRIPTION
**Pull request by Garamotte**
Migration of printer_tray to v10.0
- [x] WIP until #68 is merged (migration of the base_report_to_printer module)

### CODE REVIEW

**Comment by lasley** [comment]
@sylvain-garancher #68 is merged, so you can now rebase this.

**Comment by Garamotte** [comment]
Rebase done.

File: printer_tray/models/printing_printer.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            if tray_option['choice'] not in tray_names:
+                tray_vals = {
+                    'name': tray_option['text'],
+                    'system_name': tray_option['choice'],
+                }
+                vals_trays.append((0, 0, tray_vals))
+
+        cups_trays = set(tray_option['choice'] for tray_option
+                         in option.choices)
+        for tray in self.tray_ids:
```
Comment (lasley) [inline printer_tray/models/printing_printer.py]: `self.tray_ids.filtered(lambda r: r.system_name not in cups_trays)` & can remove conditional

File: printer_tray/models/printing_printer.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        return vals
+
+    @api.multi
+    def print_options(self, report=None, format=None, copies=1):
+        """ Hook to define Tray """
+        printing_act_obj = self.env['printing.report.xml.action']
+        options = super(PrintingPrinter, self).print_options(report, format)
+
+        # Retrieve user default values
+        tray = self.env.user.printer_tray_id
```
Comment (lasley) [inline printer_tray/models/printing_printer.py]: This should be inside of the conditional IMO, as an else to the `if report.printer_tray_id:`

File: printer_tray/models/ir_actions_report_xml.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2013-2014 Camptocamp (<http://www.camptocamp.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
```
Comment (lasley) [inline printer_tray/models/ir_actions_report_xml.py]: `from odoo import api, fields, models`

File: printer_tray/models/printing_printer.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2013-2014 Camptocamp (<http://www.camptocamp.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import errno
+import logging
+import os
+
+from odoo import models, fields, api
```
Comment (lasley) [inline printer_tray/models/printing_printer.py]: `from odoo import api, fields, models`

File: printer_tray/models/printing_report_xml_action.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2013-2014 Camptocamp (<http://www.camptocamp.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
```
Comment (lasley) [inline printer_tray/models/printing_report_xml_action.py]: `from odoo import api, fields, models`

File: printer_tray/models/printing_tray.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2013-2014 Camptocamp (<http://www.camptocamp.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields
```
Comment (lasley) [inline printer_tray/models/printing_tray.py]: `from odoo import fields, models`

File: printer_tray/models/res_users.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2013-2014 Camptocamp (<http://www.camptocamp.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
```
Comment (lasley) [inline printer_tray/models/res_users.py]: `from odoo import api, fields, models`

**Review by lasley — CHANGES_REQUESTED** [review]
Thanks @sylvain-garancher, comments inline

**Comment by Garamotte** [comment]
@lasley Done.
I took the liberty to refactor the `_prepare_update_from_cups` method.

**Review by lasley — APPROVED** [review]

**Review by moylop260 — APPROVED** [review]

**Comment by Garamotte** [comment]
I added some tests to fix the tiny coverage drop... and I think I found a bug in the rare untested lines.

@lasley @moylop260 Do you confirm that the new behaviour is what was expected ?

**Review by lasley — APPROVED** [review]
Yup that nails it, good catch @sylvain-garancher

**Comment by pedrobaeza** [comment]
Please squash OCA Transbot commits together

**Comment by Garamotte** [comment]
@pedrobaeza Done. I included @sbidoul commit that removed the en.po file added by one of the OCA transbot's squashed commits.

I also rebased on 10.0 and squashed my 3 last commits (migration, review fixes and new unit tests).

**Comment by pedrobaeza** [comment]
Thanks

**Pull request merged**